contrib/charts/cert-manager/templates/hook_apply_issuers.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.clusterIssuer.email }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: apply-issuers
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: apply-issuers
+      namespace: {{ .Release.Namespace }}
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: apply-issuers
+        image: claranet/gcloud-kubectl-docker
```

Comment:
:-1: on using a random docker image on `:latest` with an `Always` image pull policy.

That image appears to be 180MB as well, which is surprisingly large given the [official hyperkube image](https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/hyperkube-amd64@sha256:177c6b1cd19394e1b8a33988c53d1f042fdac8d5223fce5912b97cfa966c212b/details/info), which contains kubectl and every other k8s component, is only 170 MB.

I'd personally be in favour of using `gcr.io/google_containers/hyperkube-amd64:v1.9.4` here instead. It has `kubectl` on the path, so that should be fine. We should also move the pull policy to a more conservative one.

Another alternative, if we want it to be even bigger, is the 700MB [cloud-sdk image](https://hub.docker.com/r/google/cloud-sdk/) which contains kubectl as well.

Reply:
One other alternative would actually be to re-use the `controller` image we already have, and add an additional commandline or entrypoint which, using go code in this repo, creates issuers.

In fact, thinking about it more, I think a more elegant solution to the create problem is to not model it in helm/kubectl, but to instead model it as a confiugration for cert-manager itself.

Helm could pass an `--autocreate-cluster-issuer $email` flag to cert-manager, or could create a configmap which cert-manager uses to find the email and mark whether it's the first run.

I admit this doesn't solve deleting them at the end, but I think that it's a much nicer approach for the creation process for numerous reasons.

Reply:
Good catch, updated using gcr.io/google_containers/hyperkube-amd64:v1.9.4